bookwyrm/tests/lists_stream/test_signals.py

```diff
@@ -0,0 +1,111 @@
+""" testing lists_stream """
+from unittest.mock import patch
+from django.test import TestCase
+from bookwyrm import lists_stream, models
+
+
+@patch("bookwyrm.models.activitypub_mixin.broadcast_task.apply_async")
+class ListsStreamSignals(TestCase):
+    """using redis to build activity streams"""
+
+    def setUp(self):
+        """use a test csv"""
+        with patch("bookwyrm.suggested_users.rerank_suggestions_task.delay"), patch(
+            "bookwyrm.activitystreams.populate_stream_task.delay"
+        ), patch("bookwyrm.lists_stream.populate_lists_task.delay"):
+            self.local_user = models.User.objects.create_user(
+                "mouse", "mouse@mouse.mouse", "password", local=True, localname="mouse"
+            )
+            self.another_user = models.User.objects.create_user(
+                "fish", "fish@fish.fish", "password", local=True, localname="fish"
+            )
+        with patch("bookwyrm.models.user.set_remote_server.delay"):
+            self.remote_user = models.User.objects.create_user(
+                "rat",
+                "rat@rat.com",
+                "ratword",
+                local=False,
+                remote_id="https://example.com/users/rat",
+                inbox="https://example.com/users/rat/inbox",
+                outbox="https://example.com/users/rat/outbox",
+            )
+
+    def test_add_list_on_create(self, _):
+        """a new lists has entered"""
+        book_list = models.List.objects.create(
+            user=self.remote_user, name="hi", privacy="public"
+        )
+        with patch("bookwyrm.lists_stream.add_list_task.delay") as mock:
+            lists_stream.add_list_on_create_command(book_list)
```

Comment:
Hmm. So if I'm reading correctly, this doesn't really test that the post_save signal is doing anything, since we're just manually calling what the signal handler _would_ call here. Do we need to mock something so that `on_transaction` fires instead?

Reply:
Yeah, it isn't testing the function it says it is. I'm not sure how to mock (or just complete) the transaction, though